rapids/rapids.sh

```diff
@@ -39,26 +49,59 @@ function execute_with_retries() {
 
   if [[ "${CUDA_VERSION}" == "10.0" ]]; then
     local -r cudf_cuda_version="10"
   else
     local -r cudf_cuda_version="${CUDA_VERSION//\./-}"
   fi
 
+  if [[ "${SPARK_VERSION}" == "3"* ]]; then
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+      "${nvidia_repo_url}/xgboost4j-spark_${SPARK_VERSION}/${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}/xgboost4j-spark_${SPARK_VERSION}-${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}.jar" \
+      -P /usr/lib/spark/jars/
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+      "${nvidia_repo_url}/xgboost4j_${SPARK_VERSION}/${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}/xgboost4j_${SPARK_VERSION}-${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}.jar" \
+      -P /usr/lib/spark/jars/
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+      "${nvidia_repo_url}/rapids-4-spark_2.12/${SPARK_RAPIDS_VERSION}/rapids-4-spark_2.12-${SPARK_RAPIDS_VERSION}.jar" \
+      -P /usr/lib/spark/jars/
+  else
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+      "${rapids_repo_url}/xgboost4j-spark_${SPARK_VERSION}/${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}/xgboost4j-spark_${SPARK_VERSION}-${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}.jar" \
+      -P /usr/lib/spark/jars/
+    wget -nv --timeout=30 --tries=5 --retry-connrefused \
+      "${rapids_repo_url}/xgboost4j_${SPARK_VERSION}/${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}/xgboost4j_${SPARK_VERSION}-${XGBOOST_VERSION}-${SPARK_RAPIDS_VERSION}.jar" \
+      -P /usr/lib/spark/jars/
+  fi
   wget -nv --timeout=30 --tries=5 --retry-connrefused \
-    "${repo_url}/xgboost4j-spark_${RAPIDS_SPARK_VERSION}/${RAPIDS_VERSION}/xgboost4j-spark_${RAPIDS_SPARK_VERSION}-${RAPIDS_VERSION}.jar" \
-    -P /usr/lib/spark/jars/
-  wget -nv --timeout=30 --tries=5 --retry-connrefused \
-    "${repo_url}/xgboost4j_${RAPIDS_SPARK_VERSION}/${RAPIDS_VERSION}/xgboost4j_${RAPIDS_SPARK_VERSION}-${RAPIDS_VERSION}.jar" \
-    -P /usr/lib/spark/jars/
-  wget -nv --timeout=30 --tries=5 --retry-connrefused \
-    "${repo_url}/cudf/${CUDF_VERSION}/cudf-${CUDF_VERSION}-cuda${cudf_cuda_version}.jar" \
+    "${rapids_repo_url}/cudf/${CUDF_VERSION}/cudf-${CUDF_VERSION}-cuda${cudf_cuda_version}.jar" \
     -P /usr/lib/spark/jars/
 }
 
+function configure_spark() {
+  cat >>${SPARK_CONF_DIR}/spark-defaults.conf <<EOF
+
+###### BEGIN : NVIDIA GPU specific properties ######
+spark.rapids.sql.concurrentGpuTasks=2
+spark.executor.resource.gpu.amount=1
+spark.rapids.memory.pinnedPool.size=2G
+spark.executor.memoryOverhead=2G
+spark.plugins=com.nvidia.spark.SQLPlugin
+spark.executor.extraJavaOptions='-Dai.rapids.cudf.prefer-pinned=true'
+spark.locality.wait=0s
+spark.executor.resource.gpu.discoveryScript=/usr/lib/spark/scripts/gpu/getGpusResources.sh
+spark.sql.shuffle.partitions=48
+spark.sql.files.maxPartitionBytes=512m
+spark.dynamicAllocation.enabled=false
+spark.shuffle.service.enabled=false
+###### END   : NVIDIA GPU specific properties ######
+EOF
+}
+
```

Comment:
Seems like we'll need to differentiate between config for Spark 2 vs Spark 3 here

Reply:
correct, let me make a few changes.

Reply:
I have addressed this issued in my latest change. I have tested on spark 2.x images and spark 3 it is working as expected. My problem is mainly I have trouble to run test cases as it recreated the same bucket over and over then hit error. I might have the wrong setup. Would appreciate if you could help me to run the test case again. @bradmiro

Reply:
Thanks for addressing this, I'll test this now. 

As for your issues, can you see if you have a bucket `gs://dataproc-init-actions-test-<PROJECT-ID>` and delete it?